src/lib.rs

```diff
@@ -287,6 +304,70 @@ fn setup_bitcoind(
     Ok(bitcoind)
 }
 
+// Create an Electrum interface from a client and BDK-based wallet, and do some sanity checks.
+// If all went well, returns the interface to Electrum.
+fn setup_electrum(
+    config: &Config,
+    db: sync::Arc<sync::Mutex<dyn DatabaseInterface>>,
+) -> Result<Electrum, StartupError> {
+    let electrum_config = match config.bitcoin_backend.as_ref() {
+        Some(config::BitcoinBackend::Electrum(electrum_config)) => electrum_config,
+        _ => Err(StartupError::MissingElectrumConfig)?,
```

Comment:
nit: it could be asserted since this function is only ever called when this condition is true.